source/agora/common/TransactionPool.d

```diff
@@ -274,11 +284,50 @@ public class TransactionPool
     {
         () @trusted
         {
-            auto results = this.db.execute("SELECT val FROM tx_pool");
+            // TXs that went through the "selection" process and should be
+            // skipped
+            bool[Hash] skip_txs;
+
+            auto results = this.db.execute("SELECT key, val FROM tx_pool");
 
             foreach (ref row; results)
             {
-                auto tx = deserializeFull!Transaction(row.peek!(ubyte[])(0));
+                auto key = Hash(row.peek!(char[])(0));
+                if (key in skip_txs)
+                    continue;
+
+                auto tx = deserializeFull!Transaction(row.peek!(ubyte[])(1));
+
+                Set!Hash db_keys; // Gather double spent TX keys
+                if (this.selector && this.gatherDoubleSpentTXs(tx, db_keys))
```

Comment:
The` selector` variable has been already set to the `defaultSelector` function. So it doesn't need to check if it's null.

Reply:
L118 calls `opApply` with a null `selector`. It is intentional, to not filter any double-spend transactions while rebuilding `spenders`.